selfdrive/car/interfaces.py

```diff
@@ -140,9 +140,22 @@ def torque_from_lateral_accel_linear(lateral_accel_value, torque_params, lateral
     )
     friction = friction_interp if friction_compensation else 0.0
     return (lateral_accel_value / torque_params.latAccelFactor) + friction
+  
+  @staticmethod
+  def torque_from_lateral_accel_non_linear(lateral_accel_value, torque_params, lateral_accel_error, lateral_accel_deadzone, steering_angle, friction_compensation):
+    # The is a non-linear relationship between torque and lateral acceleration (accounting for road roll and steering friction)
+    # This function adjusts the lateral accel factor based on the steering angle
+    friction_interp = interp(
+      apply_center_deadzone(lateral_accel_error, lateral_accel_deadzone),
+      [-FRICTION_THRESHOLD, FRICTION_THRESHOLD],
+      [-torque_params.friction, torque_params.friction]
+    )
+    lat_factor = torque_params.latAccelFactor * ((abs(steering_angle) * torque_params.latAngleFactor) + 1) # This is the non-linear part. Adding 1 to avoid a zero factor seems apropriate
+    friction = friction_interp if friction_compensation else 0.0
```

Comment:
We're currently trying to make the Bolt good, and we think it has a non-linear response as well. Maybe something can be made generic. @nuwandavek might have some input

Reply:
Did you guys get around to trying this on the bolt?